src/components/Checkbox/Checkbox.test.tsx

```diff
@@ -0,0 +1,58 @@
+import "@testing-library/jest-dom/extend-expect"
+import React, { useState } from "react"
+import { act, fireEvent, render } from "@testing-library/react"
+
+import Checkbox, { ICheckboxProps } from "./Checkbox"
+
+const CheckboxTemplate = (props: ICheckboxProps) => {
+  const [checked, setChecked] = useState(false)
+
+  return (
+    <Checkbox
+      checked={checked}
+      onChange={() => setChecked((previous) => !previous)}
+      {...props}
+    />
+  )
+}
+
+describe("Checkbox", () => {
+  it("can be checked", () => {
+    const mockedFunction = jest.fn()
+    const { getByTestId } = render(
+      <CheckboxTemplate onChange={mockedFunction} />
+    )
+
+    const checkbox = getByTestId("checkbox")
```

Comment:
As per the testing-library suggests, I think it would be better to use getByRole here instead of getByTestId, as the checkbox role should be checkbox, so it will also test some accessibility

reference: https://testing-library.com/docs/queries/about/#priority